tests/test_crawl.py

```diff
@@ -226,3 +226,13 @@ def cb(response):
         s = dict(est[0])
         self.assertEqual(s['engine.spider.name'], crawler.spider.name)
         self.assertEqual(s['len(engine.scraper.slot.active)'], 1)
+
+    @defer.inlineCallbacks
+    def test_graceful_crawl_error_handling(self):
+        crawler = get_crawler(SimpleSpider)
+        class TestError(Exception):
+            pass
+        with mock.patch('scrapy.crawler.ExecutionEngine.open_spider') as mock_os:
```

Comment:
Thanks, it is more clear now. 

Test reads as follows: if `ExecutionEngine.open_spider` raises an error then `crawler.crawl()` should raise the same error and stop crawling. But it is not clear why `ExecutionEngine.open_spider` can raise an error - is it some protection against errors in Scrapy itself, or a protection against errors in user components? This is what made the test harder to read. Testing Engine on its own makes sense, but I think it worths adding your comment to the source code. 

A test like "Here is a faulty spider middleware, we enable it and crawl, and as a result an exception is raised and crawler is stopped" can be easier to read, but it won't check all options. Maybe we should have both. Or maybe we already have such test, sorry for noise :)